src/cat.c

```diff
@@ -323,7 +326,23 @@ static void ack_error(struct cat_object *self)
 {
         assert(self != NULL);
 
+        #if CONFIG_LIB_CAT_USRDATA
+        char *strbuf = (char *)malloc(512);
+        assert(NULL == strbuf);
+        memset(strbuf,0,ACKBUF_LEN);
```

Comment:
using dynamic allocation for this purpose is completely unnecessary. first of all, one of the major requirements is to completely avoid dynamic allocations in this library. at second, You could use variable array length, for this purpose.